test/unittest/dds/xtypes/dyndata_idl/idl_to_json_Tests.cpp

```diff
@@ -0,0 +1,443 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <nlohmann/json.hpp>
+
+#include <fastdds/dds/core/ReturnCode.hpp>
+#include <fastdds/dds/topic/TypeSupport.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/DynamicData.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/DynamicDataFactory.hpp>
+#include <fastdds/dds/xtypes/utils.hpp>
+
+#include "types/types.hpp"
+
+#include <gtest/gtest.h>
+
+using json = nlohmann::json;
+
+// ENUMERATION_BUILDER(
+//     DataTypeKind::HELLO_WORLD,
+//     );
+
+// // Crear un test en el que se testeen tipos más simples
+// TEST(idl_to_json_Tests, SimpleTypesTests)
+// {
+
+// }
+
+std::string expected_json_complete = R"({
```

Comment:
See comment below, I would have the groundtruth separated in dedicated files, each placed at the corresponding type subfolder, together with the idl file.

Reply:
Also it would be interesting to test both an empty dynamic data, and a populated one.